core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -311,7 +333,7 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {
         int result = reason != null ? reason.hashCode() : 0;
-        result = 31 * result + Long.hashCode(timestamp);
+        result = 31 * result + Long.hashCode(unassignedTimeMillis);
```

Comment:
The equals is a few lines up...
I was hesitant on using nano for equality as it varies between nodes for what we logically consider "equal" value.
Shouldn't millis be good enough here?